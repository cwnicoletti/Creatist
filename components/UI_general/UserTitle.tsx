import React, { useState } from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { AnimatedGradient } from "../custom/AnimatedGradient/AnimatedGradient";

const UserTitle = (props) => {
  const [imageIsLoading, setImageIsLoading] = useState(true);
  const [greyColorValues, setGreyColorValues] = useState([
    "rgba(50,50,50,1)",
    "rgba(0,0,0,1)",
  ]);
  const sourceImg: string | any = props.imgSource
    ? { uri: props.imgSource }
    : require("../../assets/default-profile-icon.jpg");

  return (
    <View style={styles.container}>
      {imageIsLoading && (
        <AnimatedGradient
          style={styles.gradient}
          colors={greyColorValues}
          start={{ x: 0, y: 0 }}
          end={{ x: 1, y: 0 }}
        />
      )}
      <Image
        style={{ ...styles.showCaseLocalImage, ...props.style }}
        source={sourceImg}
        onLoadStart={() => {
          setGreyColorValues(["rgba(0,0,0,1)", "rgba(50,50,50,1)"]);
        }}
        onLoadEnd={() => {
          setImageIsLoading(false);
        }}
      />
      <Text style={props.fullnameStyle}>{props.fullname}</Text>
      {props.jobTitle ? (
        <Text style={props.jobTitleStyle}>{props.jobTitle}</Text>
      ) : null}
      {props.username ? (
        <Text style={props.usernameStyle}>{props.username}</Text>
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    margin: 10,
  },

  gradient: {
    top: 0,
    height: 100,
    width: 100,
    borderRadius: 100 / 2,
    position: "absolute",
    zIndex: 3,
  },

  showCaseLocalImage: {
    height: 100,
    width: 100,
    borderRadius: 100 / 2,
  },
});

export default UserTitle;
