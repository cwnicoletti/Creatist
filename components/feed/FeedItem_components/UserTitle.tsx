import React from "react";
import { StyleSheet, Text, View } from "react-native";

const UserTitle = (props) => {
  return (
    <View
      style={{
        flexDirection: "column",
        justifyContent: "center",
      }}
    >
      <Text
        style={{
          color: props.darkModeValue ? "white" : "black",
          fontWeight: "700",
          marginLeft: 5,
          fontSize: 11,
        }}
      >
        {props.fullname}
      </Text>
      {props.jobTitle ? (
        <Text
          style={{
            color: props.darkModeValue ? "white" : "black",
            fontWeight: "500",
            fontSize: 11,
            marginLeft: 5,
          }}
        >
          {props.jobTitle}
        </Text>
      ) : null}
      <Text
        style={{
          color: "grey",
          fontSize: 10,
          marginLeft: 5,
        }}
      >
        {props.username}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({});

export default UserTitle;
