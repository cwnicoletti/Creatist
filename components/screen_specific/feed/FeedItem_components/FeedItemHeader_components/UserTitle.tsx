import React from "react";
import { StyleSheet, Text, View } from "react-native";

const UserTitle = (props) => {
  return (
    <View style={styles.container}>
      <Text
        style={{
          ...styles.fullname,
          color: props.darkModeValue ? "white" : "black",
        }}
      >
        {props.fullname}
      </Text>
      {props.jobTitle && (
        <Text
          style={{
            ...styles.jobTitle,
            color: props.darkModeValue ? "white" : "black",
          }}
        >
          {props.jobTitle}
        </Text>
      )}
      <Text style={styles.username}>{props.username}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    justifyContent: "center",
  },

  fullname: {
    fontWeight: "700",
    marginLeft: 5,
    fontSize: 11,
  },

  jobTitle: {
    fontWeight: "500",
    fontSize: 11,
    marginLeft: 5,
  },

  username: {
    color: "grey",
    fontSize: 10,
    marginLeft: 5,
  },
});

export default UserTitle;
