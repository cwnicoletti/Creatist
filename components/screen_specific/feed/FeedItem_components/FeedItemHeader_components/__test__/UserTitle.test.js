import React from "react";
import renderer from "react-test-renderer";
import UserTitle from "../UserTitle";

test("matches previous snapshot and renders correctly", () => {
  const tree = renderer.create(<UserTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});
