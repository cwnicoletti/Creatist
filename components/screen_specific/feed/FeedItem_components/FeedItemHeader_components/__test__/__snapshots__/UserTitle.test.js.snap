// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot and renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 11,
        "fontWeight": "700",
        "marginLeft": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 10,
        "marginLeft": 5,
      }
    }
  />
</View>
`;
